trestle/core/markdown/markdown_processor.py

```diff
@@ -84,3 +85,25 @@ def fetch_value_from_header(self, md_path: pathlib.Path, key: str) -> Optional[s
             value = header[key]
 
         return value
+
+    def cull_headings(self, tree: MarkdownNode, cull_list: List[str], flexible: bool = True) -> None:
```

Comment:
1) This operation is performed on the MarkdownNode, therefore please move it to MarkdownNode
2) What is `flexible` do? Add doc to this parameter

Reply:
reverted the change